app/routes/goal.py

```diff
@@ -0,0 +1,101 @@
+
+from app import db
+from flask import Blueprint,jsonify,request,abort,make_response
+from app.models.goal import Goal
+from app.models.task import Task
+import os,requests
+from .helper_function import get_model_from_id
+
+goal_bp = Blueprint("goal",__name__,url_prefix = "/goals")
+
+@goal_bp.route('',methods =['POST']) 
+def create_goal():
+    request_body = request.get_json()
+    try:
+        new_goal = Goal( title = request_body['title'])
+            
+    except:
+        return abort(make_response({"details": "Invalid data"},400))
```

Comment:
Good use of a try-except.  However I suggest providing more info to the user about what went wrong.